src/strokes.ts

```diff
@@ -130,114 +130,87 @@ export class Stroke extends Modifier {
-    // We later prepend 'stroke' to create a camelCased glyph category: 'strokeStraight' or 'strokeWiggly'.
-    // See: common_metrics.ts
-    let type = 'Straight';
-
     // Draw the stroke
     if (this.type === Stroke.Type.BRUSH_DOWN || this.type === Stroke.Type.BRUSH_UP) {
       ctx.fillRect(x + this.xShift, topY, 1, botY - topY);
     } else {
-      type = 'Wiggly';
-      if (isStaveNote(note)) {
-        for (let i = topY; i <= botY; i += lineSpace) {
-          Glyph.renderGlyph(ctx, x + this.xShift - 4, i, this.renderOptions.fontScale, 'vexWiggleArpeggioUp');
-        }
+      const lineGlyph =
+        arrow === '\ueb7c' /*arrowheadBlackDown*/ ? '\ueaa9' /*wiggleArpeggiatoUp*/ : '\ueaaa'; /*wiggleArpeggiatoDown*/
+      let txt = '';
+      const el = new Element();
+      // 2 pixels overlap of arpeggiato glyphs
+      for (; el.getWidth() < botY - topY; ) {
+        txt += lineGlyph;
+        el.setText(txt);
+        el.measureText();
+      }
+      if (
+        this.type === Stroke.Type.RASQUEDO_DOWN ||
+        this.type === Stroke.Type.ROLL_DOWN ||
+        this.type === Stroke.Type.ARPEGGIO_DIRECTIONLESS
+      ) {
+        ctx.openRotation(90, x + this.xShift, topY);
+        el.renderText(ctx, x + this.xShift, topY - el.getTextMetrics().actualBoundingBoxDescent + el.getHeight() / 2);
+        ctx.closeRotation();
+        textY = topY + el.getWidth() + 5;
       } else {
-        let i;
-        for (i = topY; i <= botY; i += 10) {
-          Glyph.renderGlyph(ctx, x + this.xShift - 4, i, this.renderOptions.fontScale, 'vexWiggleArpeggioUp');
-        }
-        if (this.type === Stroke.Type.RASQUEDO_DOWN) {
-          textY = i + 0.25 * lineSpace;
-        }
+        ctx.openRotation(-90, x + this.xShift, botY);
+        el.renderText(ctx, x + this.xShift, botY - el.getTextMetrics().actualBoundingBoxDescent + el.getHeight() / 2);
+        ctx.closeRotation();
+        textY = botY - el.getWidth() - 5;
       }
     }
 
-    if (this.type === Stroke.Type.ARPEGGIO_DIRECTIONLESS) {
-      return; // skip drawing arrow heads or text
-    }
-
     // Draw the arrow head
-    Glyph.renderGlyph(ctx, x + this.xShift + arrowShiftX, arrowY, this.renderOptions.fontScale, arrow, {
-      category: `stroke${type}.${arrow}`,
-    });
+    if (arrowY !== 0) {
+      const el = new Element();
+      el.setText(arrow);
+      el.measureText();
+      el.renderText(
+        ctx,
+        // Center the arrow head substracting its width / 2
+        x + this.xShift - el.getWidth() / 2,
+        arrowY
+      );
+    }
 
     // Draw the rasquedo "R"
     if (this.type === Stroke.Type.RASQUEDO_DOWN || this.type === Stroke.Type.RASQUEDO_UP) {
-      const textFont = Tables.lookupMetricFontInfo(`Strokes.text`);
-      ctx.save();
-      ctx.setFont(textFont);
-      ctx.fillText('R', x + textShiftX, textY);
-      ctx.restore();
+      const el = new Element('Stroke.text');
+      el.setText('R');
+      el.measureText();
```

Comment:
Later on, I wonder if it would help to have a cache for measureText() results. We can explore this after your v5 porting process is done. I feel like all the calls to setText() and measureText() will have some negative impact on performance.

Reply:
We have to look into performance, no doubt.